### ISSUE TITLE
open_run() should allow numpy integer types

### ISSUE THREAD

**Issue by takluyver** [European-XFEL/EXtra-data#31]
A researcher from HED emailed me to mention that `open_run()` fails in a confusing way when you pass it numbers from a numpy array. It should accept numpy integers like regular Python integers.

> I have a loop over ints in a numpy.array and I pass these to my function and I get this error:

```
<ipython-input-16-dd4946b73ff9> in read_images(run_no, detno, trainids)
     50     Can select only certain trainids. Default is all images and take the average."""
     51     print(propno, run_no)
---> 52     run = open_run(proposal=propno, run=run_no)
     53     #print(run.all_sources)
     54     #print(run.keys_for_source(detdaqoutput(detno)[0]))

/software/anaconda3/5.2/lib/python3.6/site-packages/extra_data/reader.py in open_run(proposal, run, data, include)
   1414     if isinstance(run, int):
   1415         run = 'r{:04d}'.format(run)
-> 1416     elif not run.startswith('r'):
   1417         run = 'r' + run.rjust(4, '0')
   1418 
```

> I believe this is because they are passed as int64 and not a simple int.

**Comment by egorsobolev** [comment]
Probably, `int(run)` or `isinstance(run, numbers.Integral)` fits better.

**Comment by takluyver** [comment]
Closed by #34.
